### PR TITLE
merge upstream archiving of students

### PR DESCRIPTION
**Pull request by marc-hanheide**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -5,6 +5,8 @@ APP_DEBUG=false
 APP_LOG_LEVEL=debug
 APP_URL=http://localhost
 URL_PREFIX=
+SESSION_SECURE_COOKIE=true
+ARCHIVE_LIMIT=2
```
Comment (marc-hanheide) [inline .env.example]: what does these do @pet1330 ?

File: app/Console/Kernel.php [hunk 2]
```diff
@@ -35,6 +35,10 @@ protected function schedule(Schedule $schedule)
         $schedule->command('reminders:starttoday')
             ->dailyAt('08:30')->pingBefore(
                 'https://hchk.io/'.env('DUE_STATUS_CHECK_KEY'));
+
+        $schedule->command('pgr:tidy-archive')->monthlyOn(1, '05:30');
+
+        $schedule->command('bouncer:clean')->monthlyOn(1, '05:35');
     }
```
Comment (marc-hanheide) [inline app/Console/Kernel.php]: what do these do?

File: composer.json [hunk 3]
```diff
@@ -17,9 +20,9 @@
         "sentry/sentry-laravel": "^0.8.0",
         "silber/bouncer": "v1.0.0-beta.5",
         "spatie/laravel-activitylog": "^2.0",
+        "spatie/laravel-backup": "5.6.*",
```
Comment (marc-hanheide) [inline composer.json]: what has this to do with it?

File: config/app.php [hunk 4]
```diff
@@ -29,6 +29,8 @@
 
     'global_message' => env('GLOBAL_MESSAGE'),
 
+    'archive_limit' => env('ARCHIVE_LIMIT', 2),
+
     /*
```
Comment (marc-hanheide) [inline config/app.php]: I'll ask again, what is the ARCHIVE_LIMIT, was is limited here?

File: tricks.md [hunk 5]
```diff
@@ -29,7 +29,7 @@ get into tinker: `php artisan tinker`
 
 ## Development
 
-* use `docker-compose -f docker-compose-devel.yml up` to fire up the development deployment
+* use `docker-compose -f docker-compose-devel.yaml up` to fire up the development deployment
```
Comment (marc-hanheide) [inline tricks.md]: :-)

File: .env.example (hunk 1 above)
Comment (marc-hanheide) [reply .env.example]: Important to know. GDPR seems to suggest that this type of record should be retained for 6 years...

Comment (pet1330) [reply .env.example]: Might be worth changing the default value to 6 then? (I created an issue for it #39)

**Pull request merged**